gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/catalog/GfxdSystemProcedures.java

```diff
@@ -2568,8 +2563,8 @@ public static String GET_SNAPSHOT_TXID() throws SQLException, StandardException
     }
     // tc commit will clear all the artifacts but will not commit actual txState
     // that should be committed in COMMIT procedure
-    tc.resetActiveTXState(true);
-    TXManagerImpl.getOrCreateTXContext().clearTXState();
+    tc.clearActiveTXState(true, true);
+    //tc.resetActiveTXState(true);
```

Comment:
With reset also, I was clearing the thread local. So, next time it shouldn't have got txState from threadLocal using  TXManagerImpl.getOrCreateTXContext().clearTXState()

_The returned txId is no longer valid, so is the caller supposed to start a transaction with the same txId later?_
Yes, next iteration may be by different connection so it needs to masqueradeAs the provided txId.

GET acts like is GET_AND_CLEAR. It lets the caller know the txId currently being used and clears the transaction controller. This is used when we finish the iteration of rowBuffer and close the connection and want to iterate over columnstore using the same tx but with different connection.